src/core/common/settings.hpp

```diff
@@ -864,6 +905,42 @@ class Settings : public SettingsBase
         bool      mIsDone;
     };
 
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+
+    /**
+     * This method saves duplicate address detection information.
+     *
+     * @param[in]   aDadInfo           A reference to a `DadInfo` structure to be saved.
+     *
+     * @retval OT_ERROR_NONE              Successfully saved duplicate address detection information in settings.
+     * @retval OT_ERROR_NOT_IMPLEMENTED   The platform does not implement settings functionality.
+     *
+     */
+    otError SaveDadInfo(const DadInfo &aDadInfo);
+
+    /**
+     * This method reads duplicate address detection information.
+     *
+     * @param[out]   aDadInfo         A reference to a `DadInfo` structure to output the read content.
+     *
+     * @retval OT_ERROR_NONE              Successfully read the duplicate address detection information.
+     * @retval OT_ERROR_NOT_FOUND         No corresponding value in the setting store.
```

Comment:
Is the `NOT_FOUND` error useful?

Reply:
`NOT_FOUND` is generally used to indicate no corresponding key storage in existing `Read` methods and  the underlying SettingDriver's `Get` method in `settings` module.
Better to keep it?